File: app/src/main/java/com/noxvision/app/hunting/weather/WeatherDtos.kt
package com.noxvision.app.hunting.weather

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class WeatherResponse(
    val main: Main,
    val wind: Wind,
    val clouds: Clouds,
    val sys: Sys,
    val weather: List<WeatherCondition>,
    val visibility: Int? = null
)

@Serializable
data class Main(
    val temp: Double,
    @SerialName("feels_like")
    val feelsLike: Double,
    val humidity: Int,
    val pressure: Int
)

@Serializable
data class Wind(
    val speed: Double,
    val deg: Int? = null,
    val gust: Double? = null
)

@Serializable
data class Clouds(
    val all: Int
)

@Serializable
data class Sys(
    val sunrise: Long,
    val sunset: Long
)

@Serializable
data class WeatherCondition(
    val description: String,
    val icon: String
)

File: NoxVision-iOS/NoxVision/Hunting/Weather/WeatherDtos.swift
import Foundation

struct WeatherResponse: Codable {
    let main: WeatherMain
    let wind: WeatherWind
    let clouds: WeatherClouds
    let sys: WeatherSys
    let weather: [WeatherCondition]
    let visibility: Int?
}

struct WeatherMain: Codable {
    let temp: Double
    let feelsLike: Double
    let humidity: Int
    let pressure: Int

    enum CodingKeys: String, CodingKey {
        case temp
        case feelsLike = "feels_like"
        case humidity
        case pressure
    }
}

struct WeatherWind: Codable {
    let speed: Double
    let deg: Int?
    let gust: Double?
}

struct WeatherClouds: Codable {
    let all: Int
}

struct WeatherSys: Codable {
    let sunrise: Int
    let sunset: Int
}

struct WeatherCondition: Codable {
    let description: String
    let icon: String
}
